src/middlewared/middlewared/service/core_service.py

```diff
@@ -881,14 +881,12 @@ def _cli_args_descriptions(self, doc, names):
             for k, v in descriptions.items()
         }
 
-    @no_auth_required
-    @api_method(CoreSetOptionsArgs, CoreSetOptionsResult, rate_limit=False)
+    @api_method(CoreSetOptionsArgs, CoreSetOptionsResult, authentication_required=False, rate_limit=False)
     @pass_app()
     async def set_options(self, app, options):
```

Comment:
@yocalebo this method is called by `truenas_api_client` to set various connection options (currently, it's only `py_exceptions`). It is called immediately on connection, before any authentication is performed. It can be safely marked as `authentication_required=False` since it only affects the current connection.

Reply:
okay